### ISSUE TITLE
Restart all threads leaves session in input mode

### ISSUE THREAD

**Issue by e9gille** [Dyalog/ride#644]
### Describe the issue you are having
Having issued a weak interrupt and then restart all threads, the session is left in input mode when it shouldn't.

### Did you connect to an already running interpreter or start the interpreter from RIDE?
Start an interpreter

### How do you reproduce the issue?
Define following function, run it then issue a weak interrupt. The tracer window appears. Click on double arrow to restart all threads.
```
     ∇ form
[1]   
[2]    'f'⎕WC'Form'
[3]    :While 1
[4]        ⎕DQ'f'
[5]    :EndWhile
     ∇

```
### Paste the contents of Help → About (Shift+F1)
```
IDE:
  Version: 4.4.3472
  Platform: Win32
  Date: 2021-01-28 07:12:15 +0100
  Git commit: 7e4f674635c8b1d35d20d35a6f341374c6ec205d
  Preferences:{
    "editWins":"{\"width\":800,\"height\":600,\"x\":200,\"y\":200,\"ox\":10,\"oy\":10}",
    "floatSingle":"0",
    "floating":"1",
    "kbdLocale":"en_GB",
    "lbar":"0",
    "selectedExe":"C:\\Program Files\\Dyalog\\Dyalog APL-64 17.1 Unicode\\dyalog.exe",
    "zoom":"-1",
    "pfkeys":"[\"\",\"\",\"<MA>\",\"\",\"\",\"\",\"\",\"\",\"\",\"\",\"\",\"\",\"\",\"\",\"\",\"\",\"\",\"\",\"\",\"\",\"\",\"\",\"\",\"\",\"\",\"\",\"\",\"\",\"\",\"\",\"\",\"\",\"\",\"\",\"\",\"\",\"\",\"\",\"\",\"\",\"\",\"\",\"\",\"\",\"\",\"\",\"\",\"\",\"\"]",
    "sbar":"0"
  }

Interpreter:
  Version: 18.0.39845
  Platform: Windows-64
  Edition: Unicode/64
  Date: Jan 18 2021 at 20:06:50

```

**Comment by mkromberg** [comment]
I also noticed yesterday, testing #627 with RIDE 4.4.3474 and Dyalog 18.1.39916, that the cursor sometimes remains locked at the left margin if threads are running.

**Comment by e9gille** [comment]
Looks like this is now fixed (tested in 18.0.40044)
